src/Microsoft.IdentityModel.JsonWebTokens/JwtTokenUtilities.cs

```diff
@@ -343,16 +344,65 @@ internal static string DecompressToken(byte[] tokenBytes, string algorithm, int
             }
         }
 
-        private static void ValidateDecryption(JwtTokenDecryptionParameters decryptionParameters, bool decryptionSucceeded, bool algorithmNotSupportedByCryptoProvider, StringBuilder exceptionStrings, StringBuilder keysAttempted)
+        private static void ValidateDecryption(JwtTokenDecryptionParameters decryptionParameters,
+            bool decryptionSucceeded,
+            bool algorithmNotSupportedByCryptoProvider,
+            StringBuilder exceptionStrings,
+            StringBuilder keysAttempted)
         {
-            if (!decryptionSucceeded && keysAttempted is not null)
-                throw LogHelper.LogExceptionMessage(new SecurityTokenDecryptionFailedException(LogHelper.FormatInvariant(TokenLogMessages.IDX10603, keysAttempted, (object)exceptionStrings ?? "", LogHelper.MarkAsSecurityArtifact(decryptionParameters.EncodedToken, SafeLogJwtToken))));
+            ExceptionDetail exceptionDetail = ValidateDecryption(
+                decryptionParameters,
+                decryptionSucceeded,
+                algorithmNotSupportedByCryptoProvider,
+                exceptionStrings,
+                keysAttempted,
+                null);
+
+            if (exceptionDetail is not null)
+                throw LogHelper.LogExceptionMessage(exceptionDetail.GetException());
+        }
 
-            if (!decryptionSucceeded && algorithmNotSupportedByCryptoProvider)
-                throw LogHelper.LogExceptionMessage(new SecurityTokenDecryptionFailedException(LogHelper.FormatInvariant(TokenLogMessages.IDX10619, LogHelper.MarkAsNonPII(decryptionParameters.Alg), LogHelper.MarkAsNonPII(decryptionParameters.Enc))));
+        private static ExceptionDetail ValidateDecryption(
+            JwtTokenDecryptionParameters decryptionParameters,
+            bool decryptionSucceeded,
+            bool algorithmNotSupportedByCryptoProvider,
+            StringBuilder exceptionStrings,
+            StringBuilder keysAttempted,
+#pragma warning disable CA1801 // Review unused parameters
```

Comment:
can callContext be removed?

Reply:
@jennyf19 we will be using this to hook up logging in the future.